tests.py

```diff
@@ -28,37 +31,39 @@ def run_vsm(self, name, input_data, expected_output, use_initial=True):
         process = Popen(cmd, stdin=PIPE, stdout=PIPE)
 
         output, _ = process.communicate(input=input_data.encode(), timeout=2)
+        output = output.decode()
+        print("result: {}".format(output))
+        self.assertEqual(output, expected_output)
 
-        self.assertEqual(output.decode() , expected_output)
```

Comment:
You don't need these changes, like I said before. All we need here is the assert.